warehouse/query-core/src/main/java/datawave/query/tld/TLDQueryIterator.java

```diff
@@ -128,13 +128,16 @@ public boolean peek(@Nullable Map.Entry<Key,String> var1) {
              * Keep any FI that is index only and part of the TLD or is not part of the TLD
              * 
              * @param k
-             * @return
+             *            a field index key
+             * @return true if the key should be kept for evaluation
              */
             @Override
             public boolean keep(Key k) {
                 boolean root = TLDEventDataFilter.isRootPointer(k);
-                DatawaveKey datawaveKey = new DatawaveKey(k);
-                return (root && getIndexOnlyFields().contains(datawaveKey.getFieldName())) || !root;
+                if (root) {
+                    return getIndexOnlyFields().contains(FiKeyUtil.getFieldString(k));
```

Comment:
Maybe this is obvious.. Is the parameter guaranteed to be a field index key?

Reply:
In this case you are guaranteed to only have field index keys.